dotnet/src/Connectors/Connectors.OpenAIV2/Services/OpenAITextEmbbedingGenerationService.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Embeddings;
+using Microsoft.SemanticKernel.Services;
+using OpenAI;
+
+namespace Microsoft.SemanticKernel.Connectors.OpenAI;
+
+/// <summary>
+/// OpenAI text embedding service.
+/// </summary>
+[Experimental("SKEXP0010")]
+public sealed class OpenAITextEmbeddingGenerationService : ITextEmbeddingGenerationService
+{
+    private readonly ClientCore _core;
+    private readonly int? _dimensions;
+
+    /// <summary>
+    /// Create an instance of the OpenAI text embedding connector
+    /// </summary>
+    /// <param name="modelId">Model name</param>
+    /// <param name="apiKey">OpenAI API Key</param>
+    /// <param name="organization">OpenAI Organization Id (usually optional)</param>
```

Comment:
Curious, when is it not optional?

Reply:
I can be wrong here but I think this happens when you are using an ApiKey of a Organization not a user.